python/ray/serve/router.py

```diff
@@ -168,7 +184,14 @@ def __init__(self, controller_handle: ActorHandle):
         self.controller = controller_handle
 
         self.endpoint_policies: Dict[str, EndpointPolicy] = dict()
-        self.backend_replicas: Dict[str, ReplicaSet] = defaultdict(ReplicaSet)
+
+        class keydefaultdict(defaultdict):
+            def __missing__(self, key):
+                self[key] = self.default_factory(key)
+                return self[key]
+
+        self.backend_replicas: Dict[str, ReplicaSet] = keydefaultdict(
+            ReplicaSet)
```

Comment:
This is still a little bit hard to read. Can we do the following:
- change `self.backend_replicas` to a dict instead of defaultdict
- add `def _get_or_create_replica_set(self, backend_name)` method.
- replace the old `self.backend_replicas[backend_tag]` with `self._get_or_create...`